### ISSUE TITLE
React Mini-search stalls on unmount for large collections

### ISSUE THREAD

**Issue by mastercoms** [lucaong/react-minisearch#38]
Hi, amazing library! I've been using it on many different projects, and it has always worked well. I recently introduced a much larger document collection to my application (about 37,000 fairly unique documents) and I've been encountering a performance problem with the cleanup introduced in https://github.com/lucaong/react-minisearch/commit/c846552398f8f185d07784a37177e25d26b0aeb4 related to these lines: https://github.com/lucaong/react-minisearch/blob/862b43ce8dfe7fc98f60c208f772e9571f80b413/src/react-minisearch.tsx#L151-L153

When my search UI unmounts, it causes a long blocking stall (about 1 second).

Here's the performance trace:

<img width="2042" alt="Chrome performance trace" src="https://github.com/lucaong/react-minisearch/assets/2672245/cfd80073-aeea-42e0-af4f-e226428bf43e">

Technically, I can manage my own document collection and supply `[]` as the documents parameter to skip over all of this, but I thought it would be good to raise this.

**Comment by lucaong** [comment]
Hi @mastercoms , thanks for the kind words :)

Regarding your issue, if you want to remove all documents, I suggest to call `removeAll` without arguments, instead of parsing the full document collection.

When calling `removeAll` passing an array, the method loops through all the given documents and removes them from the index, which is a rather expensive operation. Calling it with no argument instead, removes all documents by simply re-initializing the whole index as empty, which is a much faster constant time operation.

**Comment by mastercoms** [comment]
Thanks, this is what I do now:

```jsx
const { search, searchResults, addAll, removeAll, isIndexing } =
    useMiniSearch([], options);

  useEffect(() => {
    removeAll();
    addAll(data);

    return () => {
      removeAll();
    };
  }, [data]);
```

But I think as a suggestion to avoid this performance penalty, either this workaround should be documented or an option given in useMiniSearch to avoid the `removeAll(documents)` on unmount.

**Comment by lucaong** [comment]
@mastercoms glad to read that my suggestion helps :)

In general (but I do not know the details of your specific case), it is not necessary to clear the index upon unmount. The index is stored in a ref, so when the component unmounts it can be garbage collected.

**Comment by mastercoms** [comment]
I'm saying that this is being done by the library, introduced in the above commit and code I linked in the original post.

**Comment by lucaong** [comment]
Oooh! You are totally right! Sorry I misunderstood, I will fix this

**Comment by lucaong** [comment]
@mastercoms I just released `v6.0.3` which fixes this issue. Thank again!
